underpants.js

```diff
@@ -243,6 +411,31 @@ var _ = {};
 *   _.every([1,2,3], function(e){return e % 2 === 0}) -> false
 */
 
+// accepts a collection and a func
+// call function for every elem of collection with following parameters:
+//      if array, current elem, its index, collection
+//      if object, current val, its key, collection
+// if all elements are true, return true
+// if any are false, return false
+// if no function, return true if all truthy, otherwise false
+
+_.every = (col, func) => {
+    let hasFalse = false;
+    if (_.typeOf(func) !== "function") {
+        _.each(col, function (elem, index, collection) {
+            if (!(elem)) {
+                hasFalse = true;
+            }
+        })
+    } else {
+        _.each(col, function (elem, index, collection) {
+            if (!(func(elem, index, collection))) {
+                hasFalse = true;
+            }
+        });
+    }
+    return (hasFalse) ? false : true;
```

Comment:
```suggestion
    let hasFalse = false;
    if (_.typeOf(func) !== "function") {
        _.each(col, function (elem, index, collection) {
            if (!(elem)) {
                hasFalse = true;
            }
        })
    } else {
        _.each(col, function (elem, index, collection) {
            if (!(func(elem, index, collection))) {
                hasFalse = true;
            }
        });
    }
    return allTrue;
```

I'd **strongly** recommend a refactor here to make your return statement easier to understand. I'd personally flip the logic - assume that all values are true (maybe an `allTrue` variable instead of `hasFalse`), and update that to be false if a falsey value is found. Then your return statement wouldn't require a ternary :)